src/main/java/frc/robot/commands/AimAtTarget.java

```diff
@@ -0,0 +1,29 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.Navigation;
+import frc.robot.subsystems.drive.Drive;
+
+public class AimAtTarget extends Command {
+    Drive driveSubsystem;
+    Navigation navSubsystem;
+    int targetId;
+
+    public AimAtTarget(Drive driveSubsystem, Navigation navSubsystem, int targetId) {
+        this.driveSubsystem = driveSubsystem;
+        this.navSubsystem = navSubsystem;
+        this.targetId = targetId;
+
+        addRequirements(driveSubsystem, navSubsystem);
+    }
+
+    @Override
+    public void initialize() {
+        navSubsystem.setDesiredTarget(targetId);
+    }
+
+    @Override
+    public void execute() {
+        driveSubsystem.aim(navSubsystem.getCurrentHeading(), navSubsystem.getTargetHeading());
```

Comment:
My preference would be for us to have a generic `setAction` function that allows us to say we want to aim, and then create a supplier that feeds heading error to the Drive subssytem from Navigation.

Then in `controlDriveTrain` we would call the `aim` function in the `AIM` state, which would use the PID to drive the supplied error to 0 via rotation commands

Reply:
Note that setAction would not set the State - it would be used to transition between states. See the main robot shooter code for details